### PR TITLE
fix(a11y): Files table roles

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

Before | After
--- | ---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/components/FilesListVirtual.vue [hunk 1]
```diff
@@ -143,8 +143,10 @@ export default Vue.extend({
 	mounted() {
 		// Make the root recycle scroller a table for proper semantics
 		const slots = this.$el.querySelectorAll('.vue-recycle-scroller__slot')
-		slots[0].setAttribute('role', 'thead')
-		slots[1].setAttribute('role', 'tfoot')
+		slots[0].classList.add('files-list__header')
```
Comment (JuliaKirschenheuter) [inline apps/files/src/components/FilesListVirtual.vue:146]: i don't have another solution right now but would it be probably possible to set this class direct to an element?

**Review by JuliaKirschenheuter — COMMENTED** [review]
To which ticket this PR belongs to?

File: apps/files/src/components/FilesListVirtual.vue (hunk 1 above)
Comment (Pytal) [reply apps/files/src/components/FilesListVirtual.vue:146]: Would have done that if possible but this programmatic solution is the only available solution as the component from `vue-virtual-scroller` does not allow us to define a custom class another way :(

**Comment by Pytal** [comment]
> To which ticket this PR belongs to?

Files2Vue a11y so would call this a followup to https://github.com/nextcloud/server/pull/36534

**Review by JuliaKirschenheuter — APPROVED** [review]

File: apps/files/src/views/FilesList.vue [hunk 2]
```diff
@@ -23,7 +23,7 @@
 	<NcAppContent v-show="!currentView?.legacy"
 		:class="{'app-content--hidden': currentView?.legacy}"
 		data-cy-files-content>
-		<div class="files-list__header">
+		<div class="files-list__crumbs">
```
Comment (skjnldsv) [inline apps/files/src/views/FilesList.vue:26]: Please no, more things than breadcrumbs are/will-be in here

File: apps/files/src/components/FilesListVirtual.vue [hunk 3]
```diff
@@ -143,8 +143,10 @@ export default Vue.extend({
 	mounted() {
 		// Make the root recycle scroller a table for proper semantics
 		const slots = this.$el.querySelectorAll('.vue-recycle-scroller__slot')
-		slots[0].setAttribute('role', 'thead')
```
Comment (skjnldsv) [inline apps/files/src/components/FilesListVirtual.vue:146]: Why is the thead not needed anymore suddenly?
The first table tr row is exactly a header, no?

https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced#the_scope_attribute
image

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Comment by Pytal** [comment]
Superseded by https://github.com/nextcloud/server/pull/39808

**Pull request closed without merging**